src/test/java/uk/gov/hmcts/reform/iacasenotificationsapi/domain/personalisation/CaseOfficerPersonalisationFactoryTest.java

```diff
@@ -83,11 +98,93 @@ public void should_create_personalisation_using_defaults_where_available() {
         assertEquals(expectedPersonalisation, actualPersonalisation);
     }
 
+    @Test
+    public void should_create_personalisation_for_listed_case_notification() {
+
+        final Map<String, String> expectedPersonalisation =
+            ImmutableMap
+                .<String, String>builder()
+                .put("Appeal Ref Number", appealReferenceNumber)
+                .put("Listing Ref Number", ariaListingReference)
+                .put("Hearing Date", extractedHearingDateFormatted)
+                .put("Hearing Time", extractedHearingTime)
+                .put("Hearing Centre Address", listCaseHearingCentreAddress)
+                .build();
+
+        Map<String, String> actualPersonalisation =
+            caseOfficerPersonalisationFactory.createListedCase(asylumCase);
+
+        verify(stringProvider, times(1)).get("hearingCentreAddress", "taylorHouse");
+
+        assertEquals(expectedPersonalisation, actualPersonalisation);
+    }
+
+    @Test
+    public void should_throw_when_list_case_hearing_centre_not_present() {
+
+        when(asylumCase.read(LIST_CASE_HEARING_CENTRE, HearingCentre.class)).thenReturn(Optional.empty());
+
+        assertThatThrownBy(() -> caseOfficerPersonalisationFactory.createListedCase(asylumCase))
+            .hasMessage("listCaseHearingCentre is not present")
+            .isExactlyInstanceOf(IllegalStateException.class);
+
+        verify(stringProvider, times(0)).get("hearingCentreAddress", "taylorHouse");
+    }
+
+    @Test(expected = DateTimeParseException.class)
+    public void should_throw_when_valid_ISO_8601_date_not_present() {
+
+        caseOfficerPersonalisationFactory.extractHearingDate(invalidISO8601HearingDate);
```

Comment:
Maybe I missed it somewhere, but is there a positive test for extracting the Date and Time somewhere?

Reply:
I'm believe this is covered in this test Ravi:

public void should_create_personalisation_for_listed_case_notification()

Should I create an additional positive test for checking the extracted Date and Time only?